### PR TITLE
Prepare package structure

### PR DESCRIPTION
**Pull request by cicheck**
## What?
* Awfully large mr :crying_cat_face: 
* Copy-past of code we need with initial refactor.
* When reviewing, I recommend taking the following approach: ask yourself if you would know where to put the code of new features.

### CODE REVIEW

File: personalized_active_learning/datasets/base.py [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+        return self.annotations.merge(self.data)
+
+    def __init__(
+        self,
+        batch_size: int = 3000,
+        split_mode: SplitMode = SplitMode.TEXTS,
+        embeddings_type: str = "labse",  # TODO: Change to class
+        major_voting: bool = False,  # TODO: Not sure if we need that
+        test_major_voting: bool = False,  # TODO: Not sure if we need that
+        folds_num: int = 10,  # TODO: Can be obtained from data?
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/base.py:193]: **[Comment]** I think that `folds_num` should be a experiment's parameter - like it is now.

File: personalized_active_learning/datasets/base.py [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+        self.folds_num = folds_num
+        self.past_annotations_limit = past_annotations_limit
+        self.use_cuda = use_cuda
+
+        self._test_fold_index = test_fold_index
+        self.use_finetuned_embeddings = use_finetuned_embeddings
+        self.min_annotations_per_user_in_fold = min_annotations_per_user_in_fold
+
+        self.split_sizes = (
+            split_sizes if split_sizes is not None else [0.55, 0.15, 0.15, 0.15]
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/base.py:242]: [**Minor**] I would move default value of `split_sizes` to global config (yaml, configparser or something)

File: personalized_active_learning/datasets/base.py [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+
+        use_cuda = self.use_cuda and torch.cuda.is_available()
+
+        create_embeddings(
+            texts, embeddings_path, model_name=model_name, use_cuda=use_cuda
+        )
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        annotations = self.annotations
+        self._original_annotations = annotations.copy()
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/base.py:269]: [**Cosmetic**] 
```python
self._original_annotations = self.annotations.copy()
```

File: personalized_active_learning/datasets/base.py [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+
+        embeddings = []
+        for text_id in range(len(text_idx_to_emb.keys())):
+            embeddings.append(text_idx_to_emb[text_id])
+
+        embeddings = np.array(embeddings)
+
+        assert len(self.data.index) == len(embeddings)
+
+        self.text_embeddings = torch.tensor(embeddings)
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/base.py:305]: [**TODO**] Remember to move it to embeddings class too.

File: personalized_active_learning/datasets/base.py [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+        ]
+
+        annotations = self.annotations
+        annotation_columns = self.annotation_columns
+        folds_num = self.folds_num
+        min_class_annotations = 1
+
+        annotators_to_filter = set()
+
+        for annotation_column in annotation_columns:
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/base.py:610]: [**Cosmetic**] 
```python
for annotation_column in self.annotation_columns:
```

File: personalized_active_learning/datasets/base.py [hunk 6]
```diff
@@ -0,0 +1,633 @@
...
+            annotation_counts.annotation_number < min_annotations
+        ].annotator_id.unique()
+
+        self.annotations = self.annotations[
+            ~self.annotations.annotator_id.isin(annotators_to_ignore)
+        ]
+
+        annotations = self.annotations
+        annotation_columns = self.annotation_columns
+        folds_num = self.folds_num
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/base.py:605]: [**Cosmetic**] Delete

File: personalized_active_learning/datasets/base.py [hunk 7]
```diff
@@ -0,0 +1,633 @@
...
+                annotations_per_class >= min_class_annotations
+            ]
+
+            class_fold_per_annotator = (
+                annotations_per_class.reset_index().annotator_id.value_counts()
+            )
+
+            annotators_to_filter.update(
+                class_fold_per_annotator[
+                    class_fold_per_annotator < folds_num * class_dim
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/base.py:627]: [**Cosmetic**] 
```python
class_fold_per_annotator < self.folds_num * class_dim
```

File: personalized_active_learning/datasets/types.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import Tuple
+
+import numpy as np
+import torch
+
+
+@dataclass()
+class TextFeaturesBatch:
```
Comment (kaszycaO) [inline personalized_active_learning/datasets/types.py:9]: [**Question**] Could you add more specific docstring with information: How to add new definitions and what are the use cases? Honestly, I am little confused here (and I don't remember how to use it :) )

**Review by kaszycaO — COMMENTED** [review]
Generally I like new changes and I know how to add new functionalities (like AL strategies, acquisition functions, pretext tasks) to the package.

File: personalized_active_learning/datasets/base.py (hunk 1 above)
Comment (cicheck) [reply personalized_active_learning/datasets/base.py:193]: Right. Removed comment!

File: personalized_active_learning/datasets/base.py (hunk 3 above)
Comment (cicheck) [reply personalized_active_learning/datasets/base.py:269]: Sure, changed it.

File: personalized_active_learning/datasets/base.py (hunk 5 above)
Comment (cicheck) [reply personalized_active_learning/datasets/base.py:610]: Sure, done it!

File: personalized_active_learning/datasets/base.py (hunk 6 above)
Comment (cicheck) [reply personalized_active_learning/datasets/base.py:605]: Done, nice catch.

File: personalized_active_learning/datasets/base.py (hunk 7 above)
Comment (cicheck) [reply personalized_active_learning/datasets/base.py:627]: Done it.

File: personalized_active_learning/datasets/types.py (hunk 8 above)
Comment (cicheck) [reply personalized_active_learning/datasets/types.py:9]: Can you elaborate? I am not sure about which `definitions` and `use cases` you have in mind. :stuck_out_tongue:

Comment (kaszycaO) [reply personalized_active_learning/datasets/types.py:9]: Just add comment what is it for :laughing:

Comment (cicheck) [reply personalized_active_learning/datasets/types.py:9]: Oh, sure xd. Added:p

**Pull request merged**